common/meta/raft-store/src/state_machine/sm.rs

```diff
@@ -90,6 +90,42 @@ const SEQ_DATABASE_META_ID: &str = "database_meta_id";
 // const TREE_META: &str = "meta";
 const TREE_STATE_MACHINE: &str = "state_machine";
 
+pub enum DataChangeType {
+    Create,
+    Update,
+    Drop,
+}
+
+/// StateMachine subscriber trait
+#[async_trait::async_trait]
+pub trait StateMachineSubscriber: Debug + Sync + Send {
+    async fn database_changed(
+        &self,
+        change_type: DataChangeType,
+        key: &DatabaseLookupKey,
+        prev: &Option<SeqV<DatabaseMeta>>,
+        meta: &Option<SeqV<DatabaseMeta>>,
+    );
```

Comment:
If possible, a subscriber may not need to be a `trait`, IMHO.

A concrete type may be easier to use.

Just an idea:
If it has to be a `trait`. A trait with type parameters will be more generalized and extensible.

```rust
trait Subscriber<K, M> {
    async fn changed(&self,
        typ: DataChangeType,
        key: K,
        prev: &Option<SeqV<M>>,
        meta: &Option<SeqV<M>>);
}
```

Reply:
> 

In this case, may be need three instance for `Subscriber ` for database\table\kv？Why these three Subscriber unify into one trait impl instance?

Reply:
> ```
> trait Subscriber<K, M> {
>     async fn changed(&self,
>         typ: DataChangeType,
>         key: K,
>         prev: &Option<SeqV<M>>,
>         meta: &Option<SeqV<M>>);
> }
> ```
> 
> In this case, may be need three instance for `Subscriber ` for database\table\kv？
> ```
> pub db_subscriber: Option<Box<dyn Subscriber<DatabaseLookupKey, DatabaseMeta>>>,
> pub table_subscriber: Option<Box<dyn Subscriber<TableUpdateKey, TableMeta>>>,
> pub kv_subscriber: Option<Box<dyn Subscriber<str, SeqV>>>,
> ```
> 
> Why these three Subscriber unify into one trait impl instance?

Right, it's not that obvious in this way.